packages/providers/src.ts/formatter.ts

```diff
@@ -152,32 +154,49 @@ export class Formatter {
         formats.blockWithTransactions.transactions = Formatter.allowNull(Formatter.arrayOf(this.transactionResponse.bind(this)));
 
         formats.filter = {
-            fromBlock: Formatter.allowNull(blockTag, undefined),
-            toBlock: Formatter.allowNull(blockTag, undefined),
-            blockHash: Formatter.allowNull(hash, undefined),
+            fromTimestamp: Formatter.allowNull(timestamp, undefined),
+            toTimestamp: Formatter.allowNull(timestamp, undefined),
+            blockHash: Formatter.allowNull(hash_48, undefined),
             address: Formatter.allowNull(address, undefined),
             topics: Formatter.allowNull(this.topics.bind(this), undefined),
         };
 
         formats.filterLog = {
-            blockNumber: Formatter.allowNull(number),
-            blockHash: Formatter.allowNull(hash),
-            transactionIndex: number,
-
-            removed: Formatter.allowNull(this.boolean.bind(this)),
-
+            timestamp: timestamp,
             address: address,
             data: Formatter.allowFalsish(data, "0x"),
-
-            topics: Formatter.arrayOf(hash),
-
-            transactionHash: hash,
+            topics: Formatter.arrayOf(hash_32),
+            transactionHash: Formatter.allowNull(hash_48, undefined),
             logIndex: number,
+            transactionIndex: number
         };
 
         return formats;
     }
 
+    logsMapper(values: Array<any>): Array<Log> {
+        let logs: Log[] = [];
+        values.forEach(function (log: any) {
+            const mapped = {
+                timestamp: log.timestamp,
+                address: log.address,
+                data: log.data,
+                topics: log.topics,
+                //@ts-ignore
+                transactionHash: null, //currently not provided
+                logIndex: log.index,
+                transactionIndex: log.index,
+            };
+            logs.push(mapped);
+        });
+        return logs;
+    }
+
+    //TODO propper validation needed?
+    timestamp(value: any): string {
```

Comment:
In the hedera context, I'd validate that the transaction contains both seconds and nanoseconds. @Daniel-K-Ivanov may have other suggestions as well.

Reply:
I did a basic validation with regex, should be sufficient for now.